ast/attachment.go

```diff
@@ -123,9 +123,16 @@ func (d *AttachmentDeclaration) Doc() prettier.Doc {
 	var doc prettier.Concat
 
 	if d.Access != AccessNotSpecified {
+		var accessDoc prettier.Doc
+		if d.Access == nil {
+			accessDoc = prettier.Text("")
+		} else {
+			accessDoc = d.Access.Doc()
+		}
```

Comment:
Instead of doing the if-`nil` check everywhere manually, could we move the `nil` check to a utility method, and call that to get the `prettier.Doc` of a given element.
```suggestion
		accessDoc := getDoc(d.Access)
```
where 
```Go
func getDoc(element ast.Element)
	if element == nil {
		return prettier.Text("")
	}

	return element.Doc()
}
```
Might have to update/introduce the interface to have a `Doc` method that is available for all AST nodes/elements.

Reply:
nice!